### DISCUSSION TITLE
Looking for Additional Maintainers

### DISCUSSION THREAD

**Discussion by petergoldstein** [petergoldstein/dalli#892] [Announcements]
I'm looking for developers who are interested in serving as additional maintainers of Dalli going forward.  I'm the only currently active maintainer of Dalli and it's not ideal for a significant open source project to have a single active maintainer.  I'm especially interested in recruiting maintainers who are currently using Dalli in production environments.

As of today the code is relatively stable and there are no known defects.  There are some open discussions as to what should be included in Dalli going forward, so there's an opportunity for significant future contribution.

If you're interested, please post in the thread.  Thanks!

**Comment by jamiemccarthy** [comment]
Hi, Peter. Thanks for all your work on Dalli over the years! I'd be interested in assisting with its maintenance. I agree that as an important component of many/most ruby apps it should be well-maintained.

I'm using Dalli in prod for my current job (at Vox Media), sending our tens-of-GB memcached cluster tens-of-thousands of queries/sec. We're running a couple of custom patches I've written that are useful in odd circumstances (one of which I might write up a PR for and submit).

I've used memcached since shortly after its invention, on multiple large-ish sites. Somehow I'm even still mentioned in its [contributors page](https://github.com/memcached/memcached/blob/master/doc/CONTRIBUTORS), I think for something minor about its perl client back in the day. But I haven't really given much of anything back to the community despite memcached being such a benefit over the years. This seems like a good opportunity to do so.

Let me know what questions you may have, happy to have those conversations!

**Reply by petergoldstein** [reply]
Hi Jamie,

Thanks for stepping up.  I appreciate it.

As a first step I'm going to invite you as a collaborator to the repository. That will give you the ability to review and merge PRs.

I'd suggest you take a look at open issues and comment as you feel appropriate. There isn't much user activity on the Dalli codebase, but feel free to comment on anything that comes in.  And if you see a PR come in that materially adds value, without adding undue complexity, feel free to merge it if it's green.  You can always '@' me if you have any questions or want my opinion.  And of course feel free to submit your own PRs.

If that all goes well, we can look at adding you on the Rubygems permissions so you can publish new gem versions.

Sound good?

**Reply by jamiemccarthy** [reply]
Sure, sounds good! I've set the repo's Watch to All Activity. And I'll start looking over the open issues tomorrow.

**Comment by jamiemccarthy** [comment]
Hi Peter. I just wanted to let you know I've been looking over the issues, and getting more familiar with the codebase. Sorry I haven't been active yet, things got a little hectic at the end of the year. I'll try to dive in more in January.

Meanwhile I made a first attempt at dockerizing CI, here https://github.com/jamiemccarthy/dalli/pull/1 . So far all the containerized servers are doing is responding with "Auth failure" -- I guess a separate memcached Dockerfile would be needed for SASL tests. But it doesn't seem impossible to get working.

**Reply by jamiemccarthy** [reply]
(January wasn't great, but I spent some time with the repo this weekend and plan to spend more time soon)
